### PR TITLE
 convert all slashes/backslashes to DIR_SEPARATOR in NormalizeSlashes()

### PR DESCRIPTION
**Pull request by rfomin**
Fixes this issue: https://github.com/fabiangreffrath/woof/issues/141#issuecomment-822072248

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
Thanks, but wouldn't it be easier to just modify `M_BaseName()` to check for both `/` and `\\`?

**Comment by rfomin** [comment]
It's easier. I just don't like the `NoramalizeSlashes()` function. I think this is a bit confusing when both DIR_SEPARATOR and slash conversion are in the same codebase.

**Comment by fabiangreffrath** [comment]
The reason is, I'd rather modify code that was added to the MBF code base anyway than modify original MBF code.

**Comment by rfomin** [comment]
Tried to fix this here https://github.com/fabiangreffrath/woof/pull/166

File: Source/w_wad.c [hunk 1]
```diff
@@ -105,25 +105,26 @@ char *AddDefaultExtension(char *path, const char *ext)
 void NormalizeSlashes(char *str)
 {
   char *p;
```
Comment (fabiangreffrath) [inline Source/w_wad.c:107]: Please init `*p = str` here.

File: Source/w_wad.c [hunk 2]
```diff
@@ -105,25 +105,26 @@ char *AddDefaultExtension(char *path, const char *ext)
...
-
+/*
```
Comment (fabiangreffrath) [inline Source/w_wad.c:108]: Please keep blank lines intact and just add the comment marker lines. Also, please add a comment why we disable this code path.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: Source/w_wad.c (hunk 1 above)
Comment (fabiangreffrath) [reply Source/w_wad.c:107]: Erm, I guess this should be `p = str + strlen(str)`, right?

**Comment by fabiangreffrath** [comment]
Hm, we could also change the function th convert all `\\` and `/` to `DIR_SEPARATOR`.

**Comment by fabiangreffrath** [comment]
https://github.com/chocolate-doom/chocolate-doom/blob/174e8e8f1377440cfbceca9331c7f6fb85718024/src/m_misc.c#L676

**Comment by rfomin** [comment]
OK, I just take Choco version then.

**Comment by fabiangreffrath** [comment]
Fine, thanks! Are the three occasions in `d_main.c` the only mentions of `'/'` and `"/"` in connection to path names?

**Comment by rfomin** [comment]
> three occasions in `d_main.c` the only mentions of `'/'` and `"/"` in connection to path names

I think so. I tried searching the entire codebase and couldn't find more.

**Comment by fabiangreffrath** [comment]
Thanks again!

**Pull request merged**